app/assets/javascripts/discourse/app/lib/user-tips.js

```diff
@@ -3,111 +3,179 @@ import { iconHTML } from "discourse-common/lib/icon-library";
 import I18n from "I18n";
 import { escape } from "pretty-text/sanitizer";
 import tippy from "tippy.js";
+import isElementInViewport from "discourse/lib/is-element-in-viewport";
 
-const instances = {};
-const queue = [];
+const TIPPY_DELAY = 500;
+
+const instancesMap = {};
+window.instancesMap = instancesMap;
+
+function destroyInstance(instance) {
+  if (instance.showTimeout) {
+    clearTimeout(instance.showTimeout);
+    instance.showTimeout = null;
+  }
+
+  if (instance.destroyTimeout) {
+    clearTimeout(instance.destroyTimeout);
+    instance.destroyTimeout = null;
+  }
+
+  instance.destroy();
+}
+
+function cancelDestroyInstance(instance) {
+  if (instance.destroyTimeout) {
+    clearTimeout(instance.destroyTimeout);
+    instance.destroyTimeout = null;
+  }
+}
+
+function showInstance(instance) {
+  if (isTesting()) {
+    instance.show();
+  } else if (!instance.showTimeout) {
+    instance.showTimeout = setTimeout(() => {
+      instance.showTimeout = undefined;
+      if (!instance.state.isDestroyed) {
+        instance.show();
+      }
+    }, TIPPY_DELAY);
+  }
+}
+
+function hideInstance(instance) {
+  clearTimeout(instance.showTimeout);
+  instance.showTimeout = null;
+  instance.hide();
+}
 
 export function showUserTip(options) {
-  hideUserTip(options.id);
+  // Find if a similar instance has been scheduled for destroying recently
+  // and cancel that
+  let instance = instancesMap[options.id];
+  if (instance) {
+    if (instance.reference === options.reference) {
+      return cancelDestroyInstance(instance);
+    } else {
+      destroyInstance(instance);
+      delete instancesMap[options.id];
+    }
+  }
 
   if (!options.reference) {
     return;
   }
 
-  if (Object.keys(instances).length > 0) {
-    return addToQueue(options);
-  }
-
-  instances[options.id] = tippy(options.reference, {
-    // Tippy must be displayed as soon as possible and not be hidden unless
-    // the user clicks on one of the two buttons.
-    showOnCreate: true,
+  instancesMap[options.id] = tippy(options.reference, {
     hideOnClick: false,
     trigger: "manual",
-    theme: "user-tips",
-    zIndex: "",
-    delay: isTesting() ? 0 : 100,
-
-    // It must be interactive to make buttons work.
-    interactive: true,
+    theme: "user-tip",
+    zIndex: "", // reset z-index
```

Comment:
I would suggest a more explicit comment here, we can easily see we are resetting zIndex here, but we are left wondering why?